Expensify/App
# [TS migration] Migrate 'Badge.js' component to TypeScript

### Details



### Fixed Issues

$ https://github.com/Expensify/App/issues/25033
PROPOSAL: N/A

### Tests

1. Check if the "environment" badge is being displayed correctly with the Expensify title.
2. Check if the wallet amount badge is being displayed correctly in the Settings.

### Offline tests

Same as above.

### QA Steps

Same as above.

### PR Author Checklist


### Screenshots/Videos

<details>
<summary>Android: Native</summary>


https://github.com/Expensify/App/assets/20051562/47b6a0c0-4d78-4016-9eb0-03a5e95b458d



</details>

<details>
<summary>Android: mWeb Chrome</summary>


https://github.com/Expensify/App/assets/20051562/579035b2-695e-41e8-a971-97e992e02940



</details>

<details>
<summary>iOS: Native</summary>


https://github.com/Expensify/App/assets/20051562/1b555019-21f3-4580-be52-7ebd31a26700



</details>

<details>
<summary>iOS: mWeb Safari</summary>


https://github.com/Expensify/App/assets/20051562/47c3ee23-3b82-4676-a5af-cd2d2fb70503



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>


https://github.com/Expensify/App/assets/20051562/609546ad-f0b7-493d-8507-5676adc3a5a1


https://github.com/Expensify/App/assets/20051562/31f62ffa-e66f-42b4-87b2-707aa30692aa



</details>

<details>
<summary>MacOS: Desktop</summary>


https://github.com/Expensify/App/assets/20051562/b5d349bc-dfc0-4722-a420-904cd15b0a09



</details>
